Update VLM Reamde for v6.1 Specifications (#2591)
mlcommons/inference: multimodal/qwen3-vl/README.md

export HF_TOKEN=<your Hugging Face token>  # Optional for public models
export HF_HOME=<path to Hugging Face cache, e.g. ~/.cache/huggingface>

docker run --runtime nvidia --gpus all \
  -p 8000:8000 \
  --ipc=host \
  --env "HUGGING_FACE_HUB_TOKEN=$HF_TOKEN" \
  -v ${HF_HOME}:/root/.cache/huggingface \
  vllm/vllm-openai:latest \
  --model ${MODEL_NAME} \
  --tensor-parallel-size 4 \
  --max-model-len=32768 \
  --async-scheduling \
  --limit-mm-per-prompt.video 0 \
  --no-enable-prefix-caching  ## Must have this flag as the rule forbids prefix caching
```

### Set up endpoints

After the server is ready to listen for requests, clone [endpoints](https://github.com/mlcommons/endpoints#) on the same node—or on any host that can reach the server over HTTP. Follow the [endpoints quick start](https://github.com/mlcommons/endpoints/tree/381d13bbd27d6d52306813a51dc4e44295222d7e#quick-start) and install with either **uv**:

```bash
git clone https://github.com/mlcommons/endpoints.git
cd endpoints
uv sync
```

or **pip** in a virtual environment:

```bash
python3.12 -m venv venv && source venv/bin/activate
pip install .
```

### Configure the benchmark

> Example configs live under [endpoints/examples/08_Qwen3-VL-235B-A22B_Example](https://github.com/mlcommons/endpoints/tree/381d13bbd27d6d52306813a51dc4e44295222d7e/examples/08_Qwen3-VL-235B-A22B_Example). Set the endpoint URL in the YAML file to match your server address and port:

Reviewer: To be explicit, we should probably specify which fields in that yaml file might be adjusted by the submitter, and which fields are frozen (in which case if a submission changed it, the submission is invalid) -> which brings up a point that maybe we need a compliance test to validate the yaml file?

Author: Updated this part. I believe the yaml config should be verified by submission checker?

Reviewer: > I believe the yaml config should be verified by submission checker?

Could you discuss during the Inference WG whose (i.e., which component's) responsibility it is to validate the yaml config? It's probably a question worth clarifying. cc @nvzhihanj

Reviewer 2: Good point, I added in Mingyuan's slide and it can be handled as part of submission checker

```yaml
endpoint_config:
  endpoints:
    - "http://localhost:8000"
```

### Run the benchmark

Launch offline or server scenarios:

```bash
uv run inference-endpoint benchmark from-config \
  -c examples/08_Qwen3-VL-235B-A22B_Example/offline_qwen3_vl_235b_a22b_shopify.yaml
```

```bash
uv run inference-endpoint benchmark from-config \

>   -c examples/08_Qwen3-VL-235B-A22B_Example/online_qwen3_vl_235b_a22b_shopify.yaml

Reviewer: ```suggestion
  -c examples/08_Qwen3-VL-235B-A22B_Example/server_qwen3_vl_235b_a22b_shopify.yaml
```
I thought that the file name was already renamed?

Author: Modified.

> ```

Reviewer: ````suggestion
Launch the offline scenario:

```bash
uv run inference-endpoint benchmark from-config \
  -c examples/08_Qwen3-VL-235B-A22B_Example/offline_qwen3_vl_235b_a22b_shopify.yaml
```

Launch the server scenario:

```bash
uv run inference-endpoint benchmark from-config \
  -c examples/08_Qwen3-VL-235B-A22B_Example/online_qwen3_vl_235b_a22b_shopify.yaml
```
````

Launch the interactive scenario:

```bash
uv run inference-endpoint benchmark from-config \
  -c examples/08_Qwen3-VL-235B-A22B_Example/interactive_qwen3_vl_235b_a22b_shopify_8k.yaml
```

> ## Compliance test

Reviewer: ```suggestion
## Compliance Test
```

    - **Interactive scenario:**
      - Category Hierarchical F1 score ≥ `0.7799`. This is the 99% recovery of `0.7878`, the mean category hierarchical F1 score across 5 runs on [the BF16 version of the model](https://huggingface.co/Qwen/Qwen3-VL-235B-A22B-Instruct). The standard deviation across those 5 runs is `0.000535724`.
  - **Server scenario:**
    - Target latency is the constraint (not Time to First Token (TTFT) or Time per Output Token (TPOT)).
    - Target latency percentile: `0.99`.
    - Target latency ≤ 12 seconds.
    - Performance sample count: `48289`.
  - **Offline scenario:**
    - Number of samples in the query ≥ `48289` (every sample in the dataset is sent to the VLM endpoint at least once).
    - Performance sample count: `48289`.
  - **Interactive scenario:**
    - Target latency is the constraint (not TTFT or TPOT).
    - Target latency percentile: `0.99`.
    - Target latency ≤ 1.5 seconds.
    - Performance sample count: `8000`.
  - Testing duration ≥ 10 minutes.
  - Sample concatenation permutation is enabled.
  - You must explicitly set `--no-enable-prefix-caching` for vLLM.

> > **MLPerf Inference v6.0 round only.** The following section and the Qwen3-VL reference under `multimodal/qwen3-vl` were maintained for the **v6.0** submission round. They are **deprecated** for newer rounds; use the current MLPerf Inference docs and repository layout for later versions.

Reviewer: ```suggestion
> [!CAUTION]
> **MLPerf Inference v6.0 round only.**
> The following sections and the Qwen3-VL reference implementation under `multimodal/qwen3-vl` were maintained for the **v6.0** submission round, and they are **deprecated** for the newer rounds. Please use the above documentation along with [mlcommons/endpoints](https://github.com/mlcommons/endpoints) for the newer rounds.
```

  -v ${HF_HOME}:/root/.cache/huggingface \
  vllm/vllm-openai:latest \
  --model ${MODEL_NAME} \
  --tensor-parallel-size 4 \
  --max-model-len=32768 \
  --async-scheduling \
  --limit-mm-per-prompt.video 0 \
  --no-enable-prefix-caching  ## Must have this flag as the rule forbids prefix caching
```

### Set up endpoints

After the server is ready to listen for requests, clone [endpoints](https://github.com/mlcommons/endpoints#) on the same node—or on any host that can reach the server over HTTP. Follow the [endpoints quick start](https://github.com/mlcommons/endpoints/tree/381d13bbd27d6d52306813a51dc4e44295222d7e#quick-start) and install with either **uv**:

```bash
git clone https://github.com/mlcommons/endpoints.git
cd endpoints
uv sync
```

or **pip** in a virtual environment:

```bash
python3.12 -m venv venv && source venv/bin/activate
pip install .
```

### Configure the benchmark

Example configs live under [endpoints/examples/08_Qwen3-VL-235B-A22B_Example](https://github.com/mlcommons/endpoints/tree/381d13bbd27d6d52306813a51dc4e44295222d7e/examples/08_Qwen3-VL-235B-A22B_Example). 

#### Fields that the submitter **should** update to match their server status:

> - Served model name:

Reviewer: ```suggestion
- Served model name (to match the actual, probably quantized, model checkpoint):
```

```yaml
model_params:
  name: "Qwen/Qwen3-VL-235B-A22B-Instruct"
```

> - Endpoints url and port:

Reviewer: ```suggestion
- The URL and port number of the endpoint:
```

```yaml
endpoint_config:
  endpoints:
    - "http://localhost:8000"
```

#### Fields that the submitter **may** customize for performance tuning:

> - Target_qps (for server and interactive mode):

Reviewer: ```suggestion
- Target QPS (for the server and interactive scenarios):
```

```yaml
  load_pattern:
    type: "poisson"
    target_qps: 6.5

> ```

Reviewer: ````suggestion
```yaml
  load_pattern:
    target_qps: 6.5
```
````
Removing `type: "poisson"` to avoid the misunderstanding that `type` could be changed.

- Client worker related settings:

```yaml
client:
    num_workers: 5
    transport:
      type: zmq
      recv_buffer_size: 16777216
      send_buffer_size: 16777216
    max_connections: 1000
    worker_initialization_timeout: 120
```

#### Fileds that the submitter **MUST NOT** change for valid results:

> - Sampling parameters that specified in scection [Reference Implementation Specification](#reference-implementation-specification)

Reviewer: ```suggestion
- Sampling parameters specified in the section [Reference Implementation Specification](#reference-implementation-specification)
```

> - Datasets (Neither performance or accuracy dataset)

Reviewer: ```suggestion
- Datasets (neither for performance evaluation nor for accuracy evaluation)
```
